src/mlpack/tests/kernel_svm_test.cpp

```diff
@@ -0,0 +1,176 @@
+/**
+ * @file tests/kernel_svm_test.cpp
+ * @author Himanshu Pathak
+ *
+ * Test the Kernel SVM class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/kernel_svm/kernel_svm.hpp>
+#include <ensmallen.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::svm;
+using namespace mlpack::distribution;
+using namespace mlpack::kernel;
+
+BOOST_AUTO_TEST_SUITE(KernelSVMTest);
+
+/**
+ * Test training of linear svm for two classes on a complex gaussian dataset
+ * using smo optimizer.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMFitIntercept)
```

Comment:
May be You can quickly migrate the test from boost to catch in this pr itself in order to resolve #2523 , that would be one less test to migrate :) . Feel free to ignore if that is hectic for you.

Reply:
Sure. :)